### DISCUSSION TITLE
Fork bitwarden clients for vaultwarden

### DISCUSSION THREAD

**Discussion by dkebler** [dani-garcia/vaultwarden#2972] [Ideas]
See 

https://vaultwarden.discourse.group/t/fork-bitwarden-clients-for-vaultwarden/2064

> By @BlackDex copied the text over:

Just getting started with vaultwarden.   The one potential long term sticking issue for me as a sysop is that it depends on bitwarden maintained clients and that there are release issues between the two. 

https://github.com/dani-garcia/vaultwarden/discussions/2944.  

I don't see this getting any better in the future and potentially becoming more problematic.  In fact being tied to their clients seems to limit vaultwarden's direction and to take up time always being sure it will work with their clients.

Thankfully those clients are open source and can be forked.
https://github.com/bitwarden/clients

So am wondering if there is enough of us willing to take on this additional code burden to fork those clients and have vaultwarden provide its own clients?

Also, looking into CLI clients there is a very good project for the command line already written in Rust that could become valutwarden's official cli client.
`https://github.com/doy/rbw`
`https://github.com/mtak/docker-rbw`

and would be great to have "official" rust cli helpers/plugins like these for ssh-agent.  `https://github.com/haipengno1/bw-key`

In this way vaultwarden could be to bitwarden as nextcloud is to owncloud.  A self contained truly "open" fork heading off in its own direction.

**Comment by RealOrangeOne** [comment]
Vaultwarden is just the server. If people want to fork the clients then they can, but there's nothing that does or needs to couple those to Vaultwarden. Even then, it's a huge maintenance burden to develop separate clients. If the clients stop being maintained, _then_ a fork makes sense, but there's no sign of that happening, and even if it did, the code is open source, so a final snapshot of the code could be used in perpetuity with the current vaultwarden client, or even continued from there.

**Reply by BlackDex** [reply]
I agree with @RealOrangeOne.
Also, i have no good knowledge on Typescript and Node etc... to work on those items effectively.
If someone wants to fork those clients, they can, as long as they do not name it Vaultwarden, that will bring more confusion to the table.

We try to keep up-to-date with the latest clients, but sometimes we have added new features for which we need to iron out some wrinkles, which takes time, and therefore we do not release a new version yet.

Also, in the current case, Bitwarden changed a lot on there clients/server which we needed to implement, and that also takes time to test and work on. Since at the moment this project is mainly maintained by 2 till 4 people with some help of a bunch of great (sometimes recurring) contributors, it just takes time.

For me it doesn't validate forking all the clients and build them with a different name.
